ant-networking/src/record_store.rs

```diff
@@ -138,7 +138,7 @@ pub struct NodeRecordStore {
     /// The configuration of the store.
     config: NodeRecordStoreConfig,
     /// Main records store remains unchanged for compatibility
-    records: HashMap<Key, (NetworkAddress, ValidationType)>,
+    records: HashMap<Key, (NetworkAddress, ValidationType, DataTypes)>,
```

Comment:
Would it be possible to ditch `ValidationType` and instead use the `DataTypes` to derive it instead? Just a nit pick tbh, will make things a bit cleaner and can be a follow up if sounds reasonable to you..

Reply:
`ValidationType` also need to hold the `content_hash` for non-chunk data.
Meanwhile, DataTypes are `public API` which supposed to keep as simple as possible.
So, don't think `merge them` is good to the public user.

Reply:
Ah sorry, overlooked that! Thanks!